ipynb/android/workloads/Android_YouTube.ipynb

```diff
@@ -0,0 +1,529 @@
+   },
+   "outputs": [],
+   "source": [
+    "def set_performance():\n",
+    "    target.cpufreq.set_all_governors('performance')\n",
+    "\n",
+    "def set_powersave():\n",
+    "    target.cpufreq.set_all_governors('powersave')\n",
+    "\n",
+    "def set_interactive():\n",
+    "    target.cpufreq.set_all_governors('interactive')\n",
+    "\n",
+    "def set_sched():\n",
+    "    target.cpufreq.set_all_governors('sched')\n",
+    "\n",
+    "def set_ondemand():\n",
+    "    target.cpufreq.set_all_governors('ondemand')\n",
+    "    \n",
+    "    for cpu in target.list_online_cpus():\n",
+    "        tunables = target.cpufreq.get_governor_tunables(cpu)\n",
+    "        target.cpufreq.set_governor_tunables(\n",
+    "            cpu,\n",
+    "            'ondemand',\n",
+    "            **{'sampling_rate' : tunables['sampling_rate_min']}\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# CPUFreq configurations to test\n",
+    "confs = {\n",
+    "    'performance' : {\n",
+    "        'label' : 'prf',\n",
+    "        'set' :  set_performance,\n",
+    "    },\n",
+    "#    'powersave' : {\n",
+    " #       'label' : 'pws',\n",
+    "  #      'set' :  set_powersave,\n",
+    "   # },\n",
+    "     'interactive' : {\n",
+    "         'label' : 'int',\n",
+    "         'set' :  set_interactive,\n",
+    "     },\n",
+    "#    'sched' : {\n",
+    " #       'label' : 'sch',\n",
+    "  #      'set' :  set_sched,\n",
+    "   # },\n",
+    "#    'ondemand' : {\n",
+    " #       'label' : 'odm',\n",
+    "  #      'set' :  set_ondemand,\n",
+    "   # }\n",
+    "}\n",
+    "\n",
+    "# The set of results for each comparison test\n",
+    "results = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def youtube_run(exp_dir, video_url, video_duration_s):\n",
+    "    # Unlock device screen (assume no password required)\n",
+    "    target.execute('input keyevent 82')\n",
+    "    # Press Back button to be sure we run the video from the start\n",
+    "    target.execute('input keyevent KEYCODE_BACK')\n",
+    "\n",
+    "    # Start YouTube video on the target device\n",
+    "    target.execute('am start -a android.intent.action.VIEW \"{}\"'.format(video_url))\n",
+    "    # Allow the activity to start\n",
+    "    sleep(3)\n",
+    "    # Reset framestats collection\n",
+    "    target.execute('dumpsys gfxinfo com.google.android.youtube reset')\n",
+    "    # Wait until the end of the video\n",
+    "    sleep(video_duration_s)\n",
+    "    \n",
+    "    # Get frame stats\n",
+    "    framestats_file = os.path.join(exp_dir, \"framestats.txt\")\n",
+    "    adb_command(target.adb_name, 'shell dumpsys gfxinfo com.google.android.youtube > {}'.format(framestats_file))\n",
```

Comment:
Maybe use a const for the format string to keep this line more short and readable, e.g.

```python
YOUTUBE_CMD = 'shell dumpsys gfxinfo com.google.android.youtube > {}'
def youtube_run(exp_dir, video_url, video_duration_s):
    ...
    ...
    adb_command(target.adb_name, YOUTUBE_CMD.format(framestats_file))